addons/account/models/account_move.py

```diff
@@ -740,22 +742,21 @@ def check_full_reconcile(self):
         In case of full reconciliation, all moves belonging to the reconciliation will belong to the same account_full_reconcile object.
         """
         # Get first all aml involved
-        part_recs = self.env['account.partial.reconcile'].search(['|', ('debit_move_id', 'in', self.ids), ('credit_move_id', 'in', self.ids)])
-        amls = self
-        todo = set(part_recs)
-        seen = set()
+        todo = self.env['account.partial.reconcile'].search_read(['|', ('debit_move_id', 'in', self.ids), ('credit_move_id', 'in', self.ids)], ['debit_move_id', 'credit_move_id'])
+        amls = self.ids
+        seen = self.env['account.partial.reconcile']
         while todo:
-            partial_rec = todo.pop()
-            seen.add(partial_rec)
-            for aml in [partial_rec.debit_move_id, partial_rec.credit_move_id]:
-                if aml not in amls:
-                    amls += aml
-                    for x in aml.matched_debit_ids | aml.matched_credit_ids:
-                        if x not in seen:
-                            todo.add(x)
-        partial_rec_ids = [x.id for x in seen]
+            aml_ids = [apr['debit_move_id'][0] for apr in todo if apr['debit_move_id']] + [apr['credit_move_id'][0] for apr in todo if apr['credit_move_id']]
+            amls += aml_ids
+            seen |= self.env['account.partial.reconcile'].browse([apr['id'] for apr in todo])
```

Comment:
according to @nseinlet , `apr` is a protected keyword that should be avoided at all costs in the code, to avoid giving buttons to anyone reading it... :see_no_evil:

scratch your head if you agree